### DISCUSSION TITLE
Does rebooting the iKinCartesianSolver requires to reboot the yarprobotinterface?

### DISCUSSION THREAD

**Discussion by S-Dafarra** [robotology/community#487] [Q&A] [answered]
Let's consider a robot with the cartesian devices activated in the robot interface, as in the iCubGenova04 case:https://github.com/robotology/robots-configuration/blob/master/iCubGenova04/icub_wbd.xml#L114-L115

With @vvasco and @vtikha we started wondering if it is necessary to reboot the ``yarprobotinterface`` every time the corresponding ``iKinCartesianSolver`` need to be rebooted.

In particular, this seemed to be necessary in the past and we had this doubt recently during a demo, so it may be worth to just have confirmation.

**Comment by pattacini** [comment, marked as answer]
The connection between the Cartesian controller and the Cartesian solver takes place once at startup. If the Cartesian solver gets stopped afterward, there is no mechanism that is responsible for establishing the connection again. This was meant to keep communication simple.

So, to answer your question, yes, you'd need to reboot `yarprobotinterface`.
Anyway, if this is somehow blocking in your context, you could always take the Cartesian controllers out of the main deployer and set up a dedicated `yarprobotinterface` deployer just for them.

This is the [relevant documentation][1].

[1]: https://robotology.github.io/robotology-documentation/doc/html/icub_cartesian_interface.html#sec_cart_dependencies

**Reply by S-Dafarra** [reply]
Thanks @pattacini, that's clear.
